pkg/storage/unified/resource/server.go

```diff
@@ -340,19 +361,23 @@ func (s *server) newEvent(ctx context.Context, user claims.AuthInfo, key *Resour
 		return nil, err
 	}
 
-	folder := obj.GetFolder()
-	if folder != "" {
-		err = s.access.CanWriteFolder(ctx, user, folder)
-		if err != nil {
-			return nil, AsErrorResult(err)
+	check.Folder = obj.GetFolder()
```

Comment:
For Create+Update the authz server needs to know if we can write the object inside this folder -- I think that is reasonable, but good to make explicit.

Lets keep things simple now -- but not sure if we need to have explicit checks when the folder value changes vs stays the same 🤔

Reply:
The update scenario is pretty interesting 🤔. How does it currently work in grafana ? Shall I be able to move/update a dashboard if I don't have access to the old folder ?

Reply:
I think:
1. if the folder changes (create or update) then you must have access to write into the new folder
2. if no-change to folder, no need to check folder permissions

we never need to check permissions on the "old" folder.  Yes this means if the folder permissions change it may no-longer be valid, but I think that is the same as changing folder permissions in drive after a file exists in it